MonoGame.Framework/Graphics/GraphicsDevice.DirectX.cs

```diff
@@ -1477,6 +1473,82 @@ private void PlatformDrawInstancedPrimitives(PrimitiveType primitiveType, int ba
             }
         }
 
+        private void PlatformGetBackBufferData<T>(Rectangle rect, T[] data, int startIndex, int count) where T : struct
```

Comment:
Hmm, I don't think that would be convenient for this part of the codebase. 
Some thoughts:
- For DirectX SharpDX provides functionality to read T's from the unmanaged memory. For OpenGL we just pass an IntPtr to the start of the managed T array and the copying/marshalling is handled by PInvoke/interop code. That's both pretty convenient and by making it more generic we'd need just as much or more code for both targets.
- If we return a byte array we'd need two copy operations, one to go from unmanaged (IntPtr) to managed memory (byte[]) and another for marshalling the byte array to a T array. Though I recall reading about the last part being done without copying with some very hacky, unsafe code. 
- In DX we can't return an IntPtr directly because of the rowPitch and because the SharpDX DataStream should be disposed inside the platform specific call. GL would be fine with an IntPtr. We do need to flip the rows of the returned data, so that would probably be done in an unsafe block to keep things cheap.

I haven't dealt with C# marshalling/interop outside of stuff I did for MG, so it's probably good to let some more experienced people look into this.